corehq/apps/geospatial/utils.py

```diff
@@ -185,3 +187,24 @@ def update_cases_owner(domain, case_id_to_owner_id, chunk_size=100):
             domain=domain,
             device_id='corehq.apps.geospatial.utils.update_cases_owners'
         )
+
+
+class CeleryTaskExistenceHelper(object):
```

Comment:
Since the method decorator will be used with the celery task, I believe it would be triggered when the celery task actually runs.  

We want to update the task state as requested as soon it is invoked even if it is just queued and not picked up.
I think this won't be possible with the decorator.